eng/WpfArcadeSdk/tools/ApiCompat.targets

```diff
@@ -0,0 +1,225 @@
+    <RunRefApiCompat>false</RunRefApiCompat>
+    <RunRefApiCompat Condition="'$(RefApiCompatNeededProjects)'!=''
+                                 and $(RefApiCompatNeededProjects.Contains('$(MSBuildProjectName)'))
+                                 and !($(MSBuildProjectName.Contains('_wpftmp')))">true</RunRefApiCompat>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Turn off default API Compat targets in favor of WPF ones. -->
+    <RunApiCompat>false</RunApiCompat>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.DotNet.ApiCompat" Version="$(MicrosoftDotNetApiCompatVersion)" GeneratePathProperty="true"
+                      Condition="'$(RunNetFrameworkApiCompat)'=='true' or '$(RunRefApiCompat)'=='true'"/>
+  </ItemGroup>
+
+  <!-- This target sets up the matching contracts to run ApiCompat from .NET Core assemblies against the .NET Framework 4.8 reference assemblies. -->
+  <Target Name="ResolveNetFrameworkApiCompatItems"
+          BeforeTargets="WpfValidateApiCompatForNetFramework"
+          Condition="'$(RunNetFrameworkApiCompat)'=='true'">
+    <PropertyGroup>
+      <ApiCompatBaseline>$(WpfApiCompatBaselineDir)$(MSBuildProjectName)-Net48.baseline.txt</ApiCompatBaseline>
+    </PropertyGroup>
+    <ItemGroup>
+      <ResolvedMatchingContract Include="$(Net48RefAssembliesDir)$(AssemblyName).dll" />
+
+      <!-- If building a hand-crafted ref assembly, compare against the current output.  Otherwise, use the ref directory (compiler generated ref). -->
+      <ResolvedImplementationAssembly Include="$(IntermediateOutputPath)$(AssemblyName).dll" Condition="!Exists('$(IntermediateOutputPath)ref\$(AssemblyName).dll')" />
+      <ResolvedImplementationAssembly Include="$(IntermediateOutputPath)ref\$(AssemblyName).dll" Condition="Exists('$(IntermediateOutputPath)ref\$(AssemblyName).dll')" />
+    </ItemGroup>
+  </Target>
+
+  <!-- This target sets up the matching contracts to run ApiCompat from .NET Core assemblies against their reference assemblies. -->
+  <Target Name="ResolveRefApiCompatItems"
+          BeforeTargets="WpfValidateApiCompatForRef"
+          Condition="'$(RunRefApiCompat)'=='true'">
+    <PropertyGroup>
+      <ApiCompatBaseline>$(WpfApiCompatBaselineDir)$(MSBuildProjectName)-ref.baseline.txt</ApiCompatBaseline>
+    </PropertyGroup>
+    <ItemGroup>
+      <ResolvedMatchingContract Include="$(OutputPath)$(AssemblyName).dll" />
+      <!-- Work backwards to find the output for a hand-crafted ref assembly. -->
+      <ResolvedImplementationAssembly Include="$(BaseOutputPath)..\$(AssemblyName)-ref\**\$(AssemblyName).dll" />
+    </ItemGroup>
+  </Target>
+
+  <!-- 
+       The below is a modified version of Arcade's ApiCompat targets file.
+       https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.ApiCompat/build/Microsoft.DotNet.ApiCompat.targets
+       
+       We use a custom version to provide greater control over which assembly pairs are being examined.
```

Comment:
What is it about Arcade's implementation that doesn't work for us?

Reply:
That's what the sentence describes, we wanted greater control over what files we pass in for both the contract and implementation.  That's pretty much it.

Reply:
This will resolve as part of adding a doc for this.  I'll point to the doc and it will have a better discussion.